### PR TITLE
New metrics to count issues by type

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-7338 SONAR-7339 SONAR-7340 new metrics to count issues by type

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 1]
```diff
@@ -42,13 +42,30 @@
...
   public static String DOMAIN_ISSUES = "Issues";
   public static String DOMAIN_GENERAL = "General";
   public static String DOMAIN_DUPLICATION = "Duplication";
   public static String DOMAIN_DESIGN = "Design";
 
   /**
+   * SonarQube Quality Model
+   * @since 5.5
+   */
+  public static String DOMAIN_MAINTAINABILITY = "Maintainability";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_MAINTAINABILITY" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MINOR Rename this field "DOMAIN_MAINTAINABILITY" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)
MAJOR Make DOMAIN_MAINTAINABILITY a static final constant or non-public and provide accessors if needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 2]
```diff
@@ -42,13 +42,30 @@
...
+   * SonarQube Quality Model
+   * @since 5.5
+   */
+  public static String DOMAIN_RELIABILITY = "Reliability";
+
+  /**
+   * SonarQube Quality Model
+   * @since 5.5
+   */
+  public static String DOMAIN_SECURITY = "Security";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_SECURITY" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MINOR Rename this field "DOMAIN_SECURITY" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)
MAJOR Make DOMAIN_SECURITY a static final constant or non-public and provide accessors if needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 3]
```diff
@@ -42,13 +42,30 @@
...
+   * SonarQube Quality Model
+   * @since 5.5
+   */
+  public static String DOMAIN_MAINTAINABILITY = "Maintainability";
+
+  /**
+   * SonarQube Quality Model
+   * @since 5.5
+   */
+  public static String DOMAIN_RELIABILITY = "Reliability";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_RELIABILITY" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MINOR Rename this field "DOMAIN_RELIABILITY" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)
MAJOR Make DOMAIN_RELIABILITY a static final constant or non-public and provide accessors if needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 6 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
LGTM

**Pull request closed without merging**